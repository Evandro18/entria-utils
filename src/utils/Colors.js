// @flow
type RGB = {
  r: number,
  g: number,
  b: number,
};
export const hexToRGBObject = (hex: string): RGB => {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Bad hex');
  }

  return {
    r: parseInt(result[1], 16),
    g: parseInt(result[2], 16),
    b: parseInt(result[3], 16),
  };
};

export const hexToRGB = (hex: string): string => {
  const rgb = hexToRGBObject(hex);
  return `rgba(${rgb.r},${rgb.g},${rgb.b})`;
};

export const hexToRGBA = (hex: string, alpha: number = 1): string => {
  const rgb = hexToRGBObject(hex);
  return `rgba(${rgb.r},${rgb.g},${rgb.b},${alpha})`;
};
