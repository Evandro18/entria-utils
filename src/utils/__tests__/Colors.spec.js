import * as Colors from '../Colors';

it('hexToRGBObject', () => {
  expect(Colors.hexToRGBObject('#ffffff')).toMatchSnapshot();
});

it('hexToRGBObject - bad hex', () => {
  function testHexToRGBObject() {
    Colors.hexToRGBObject('#fff');
  }

  expect(testHexToRGBObject).toThrowErrorMatchingSnapshot();
});

it('hexToRGB', () => {
  expect(Colors.hexToRGB('#ffffff')).toMatchSnapshot();
});

it('hexToRGBA - without alpha', () => {
  expect(Colors.hexToRGBA('#ffffff')).toMatchSnapshot();
});

it('hexToRGBA - with alpha', () => {
  expect(Colors.hexToRGBA('#ffffff', 0.5)).toMatchSnapshot();
});
