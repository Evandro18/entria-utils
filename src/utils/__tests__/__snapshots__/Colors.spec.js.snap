// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hexToRGB 1`] = `"rgba(255,255,255)"`;

exports[`hexToRGBA - with alpha 1`] = `"rgba(255,255,255,0.5)"`;

exports[`hexToRGBA - without alpha 1`] = `"rgba(255,255,255,1)"`;

exports[`hexToRGBObject - bad hex 1`] = `"Bad hex"`;

exports[`hexToRGBObject 1`] = `
Object {
  "b": 255,
  "g": 255,
  "r": 255,
}
`;
